examples/eSealing/src/utils.ts

```diff
@@ -8,23 +18,46 @@ import { E_SEALING_CONTRACT_NAME, E_SEALING_RAW_SCHEMA } from './constants';
  */
 export async function register(
     connection: WalletConnection,
+    grpcClient: ConcordiumGRPCClient,
     account: string,
     fileHashHex: string,
     index: bigint,
     subindex = 0n
 ) {
+    const contractAddress = ContractAddress.create(index, subindex);
+    const receiveName = ReceiveName.fromString(`${E_SEALING_CONTRACT_NAME}.registerFile`);
+
+    const schema = moduleSchemaFromBase64(E_SEALING_RAW_SCHEMA);
+    const serializedParameters = serializeUpdateContractParameters(
+        ContractName.fromString(E_SEALING_CONTRACT_NAME),
+        EntrypointName.fromString('registerFile'),
+        fileHashHex,
+        schema.value
+    );
+
+    const invokeResult = await grpcClient.invokeContract({
+        contract: contractAddress,
+        method: receiveName,
+        invoker: AccountAddress.fromBase58(account),
+        parameter: serializedParameters,
+    });
+
+    if (invokeResult.tag === 'failure') {
+        throw Error('Transaction would fail!');
+    }
+
     return connection.signAndSendTransaction(
         account,
         AccountTransactionType.Update,
         {
             amount: CcdAmount.fromMicroCcd(0),
-            address: ContractAddress.create(index, subindex),
-            receiveName: ReceiveName.fromString(`${E_SEALING_CONTRACT_NAME}.registerFile`),
-            maxContractExecutionEnergy: Energy.create(30000n),
+            address: contractAddress,
+            receiveName,
+            maxContractExecutionEnergy: invokeResult.usedEnergy,
```

Comment:
Adding a small margin `EPSILON_ENERGY` will help in cases where someone else uses the front end at the same time and the state of the smart contract has changed between the simulation and the execution of the transaction.
```
    const maxContractExecutionEnergy = Energy.create(dryRunResult.usedEnergy.value + EPSILON_ENERGY);

```

https://github.com/Concordium/concordium-dapp-examples/blob/main/trackAndTrace/frontend/src/track_and_trace_contract.tsx#L63

Applies to the other examples below as well.